pt/lessons/specifics/mnesia.md

```diff
@@ -0,0 +1,185 @@
+
+## Criando Tabelas
+
+A função `Mnesia.create_table/2` é usada para criar tabelas dentro do nosso banco de dados. Abaixo criamos uma tabela chamada `Person`, e em seguida, passamos uma lista de palavra-chave definindo o schema da tabela.
+
+```shell
+iex> Mnesia.create_table(Person, [attributes: [:id, :name, :job]])
+{:atomic, :ok}
+```
+Nós definimos as colunas usando os átomos `:id`, `:name`, e ':job'. Quando executamos `Mnesia.create_table/2`, ele irá retornar qualquer uma das seguinte respostas:
+
+ - `{atomic, ok}` se a função foi executada com êxito
+ - `{aborted, Reason}` se a função falhou
+
+## A Maneira Suja
+
+Primeiro de tudo, vamos olhar para a maneira suja de leitura e escrita em uma tabela no Mnesia. Isso geralmente deve ser evitado, pois o sucesso não é garantido, mas deve nos ajudar a aprender e tornar-se confortável trabalhando com Mnesia. Vamos adicionar algumas entradas para nossa tabela **Person**.
+
+```shell
+iex> Mnesia.dirty_write({Person, 1, "Seymour Skinner", "Principal"})
+:ok
+
+iex> Mnesia.dirty_write({Person, 2, "Homer Simpson", "Safety Inspector"})
+:ok
+
+iex> Mnesia.dirty_write({Person, 3, "Moe Szyslak", "Bartender"})
+:ok
+```
+
+... e para recuperar as entradas podemos usar `Mnesia.dirty_read/1`:
+
+```shell
+iex> Mnesia.dirty_read({Person, 1})
+[{Person, 1, "Seymour Skinner", "Principal"}]
+
+iex> Mnesia.dirty_read({Person, 2})
+[{Person, 2, "Homer Simpson", "Safety Inspector"}]
+
+iex> Mnesia.dirty_read({Person, 3})
+[{Person, 3, "Moe Szyslak", "Bartender"}]
+
+iex> Mnesia.dirty_read({Person, 4})
+[]
+```
+Se nós tentamos consultar um registro que não existe, Mnesia irá responder com uma lista vazia.
+
+
+## <a name="transacoes"></a> Transações
+
+Tradicionalmente nós usamos **transações** para encapsular nossas leituras no nosso banco de dados. Transações são uma parte importante para concepção de sistemas altamente distribuídos e tolerantes a falhas. Uma *transação* no Mnesia é um mecanismo através do qual uma série de operações da base de dados pode ser executada como um bloco funcional. Primeiro criamos uma função anônima, neste caso `data_to_write` e em seguida, passamos esta função para `Mnesia.transaction`.
+
+```shell
+iex> data_to_write = fn ->
+...>   Mnesia.write({Person, 4, "Marge Simpson", "home maker"})
+...>   Mnesia.write({Person, 5, "Hans Moleman", "unknown"})
+...>   Mnesia.write({Person, 6, "Monty Burns", "Businessman"})
+...>   Mnesia.write({Person, 7, "Waylon Smithers", "Executive assistant"})
+...> end
+#Function<20.54118792/0 in :erl_eval.expr/5>
+
+iex> Mnesia.transaction(data_to_write)
+{:atomic, :ok}
+```
+Com base neste mensagem de transação, podemos seguramente assumir que nós escrevemos os dados para a nossa tabela `Person`. Vamos usar  uma transação para ler a partir do banco de dados agora para ter certeza. Usaremos `Mnesia.read/1` para ler a partir do banco de dados, mais uma vez de dentro de uma função anônima.
```

Comment:
I think there's an extra whitespace between `usar` and `uma` in `Vamos usar uma transação...`